File: cpp/leetcode/071.cpp
string simplifyPath(string path) {
    string res, ch;
    vector<string> stk;
    stringstream ss(path);
    while(getline(ss, ch, '/')) {
        if (ch == "" or ch == ".") continue;
        if (ch == ".." and !stk.empty()) stk.pop_back();
        else if (ch != "..") stk.push_back(ch);
    }
    for(auto str : stk) res += "/"+str;
    return res.empty() ? "/" : res;
}

File: python/leetcode/071.py
class Solution(object):
    def simplifyPath(self, path):
        strs = path.split("/")
        res = []
        for ch in strs:
            if not ch or ch == '.': continue
            if ch == '..' and len(res) > 0: res.pop()
            elif ch != '..': res.append(ch)
        return "/" + "/".join(res)